### PR TITLE
Cylc TextMate Bundle

### PR DESCRIPTION
**Pull request by oliver-sanders**
A language bundle for:

* TextMate
* PyCharm
* WebStorm
* Sublime

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Cylc TextMate Grammar
+
+A TextMate bundle for working with [Cylc](www.cylc.org) workflow
```
Comment (kinow) [inline README.md]: Without the protocol in front of the URL, in Markdown that gets resolved to a relative URL of the current git repo

**Review by kinow — COMMENTED** [review]
Successfully installed on WebStorm. It doesn't recognize `suite.rc` as a file to be interpreted by this bundle though. Nor suite.cylc or suite.Cylc.

Looking at another bundle, it looks like this could have a `filetype` key somewhere in one of the files (grain of salt, as I never learned anything about TextMate bundles until just now :sweat_smile:): https://github.com/marko-js/marko-tmbundle/blob/2ecea1ebf5f8793954ea73b367e3857b9c10117b/Syntaxes/Marko.tmLanguage#L5

**Comment by kinow** [comment]
Diff:

```diff
diff --git a/Syntaxes/cylc.tmLanguage b/Syntaxes/cylc.tmLanguage
index a2d9292..de688f4 100644
--- a/Syntaxes/cylc.tmLanguage
+++ b/Syntaxes/cylc.tmLanguage
@@ -4,6 +4,10 @@
 <dict>
        <key>name</key>
        <string>cylc</string>
+        <key>fileTypes</key>
+        <array>
+                <string>rc</string>
+        </array>
        <key>patterns</key>
        <array>
                <dict>
```

Then removed, added back to WebStorm, and loaded a suite I used recently in a PR for cylc-admin.

image

Looking great! :rocket:

**Comment by kinow** [comment]
(random thought too: maybe there is a way to create a JS app for this syntax? If so, we could have an online syntax highlight for Cylc suites, maybe deployed to GitHub pages or Netlify; even though there are IDE and editors that support syntax, some users may want to just read some suite they have without installing anything—there are tools like this for HTML, Java, Prolog, etc)

**Comment by oliver-sanders** [comment]
> Diff:

Yes ATM you have to tell your editor to use this syntax explicitly, this needs to be fixed in the text mate grammar itself. Note that using the `.rc` extension isn't ok because it will override default `.rc` behaviour.

> maybe there is a way to create a JS app for this syntax?

Hopefully yes - https://github.com/cylc/cylc-ui/issues/144

**Comment by kinow** [comment]
>Yes ATM you have to tell your editor to use this syntax explicitly, this needs to be fixed in the text mate grammar itself. Note that using the .rc extension isn't ok because it will override default .rc behaviour.

:+1:  I will look into how to set the extension in WebStorm then (I know I can do that with Eclipse, no idea with WebStorm).

**Comment by oliver-sanders** [comment]
I think we will associate `*suite.rc` files with the grammar in a future commit.

**Comment by kinow** [comment]
> I think we will associate `*suite.rc` files with the grammar in a future commit.

That would be helpful. At the moment in WebStorm the `.rc` extension has no association, and opens with the default text editor. But `*suite.rc` would probably work for WebStorm and for users that have already .rc associated with other editors/bundles.

**Review by kinow — APPROVED** [review]
Looks good to me. The README has that link that is relative to GitHub, instead of https://cylc.github.io. But no biggie IMO. And works OK on WebStorm. LGTM

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**